crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3471,6 +3471,12 @@ impl<'db> TypeInferenceBuilder<'db> {
                 Type::Todo
             }
 
+            // TODO PEP-604 unions
+            ast::Expr::BinOp(binary) => {
+                self.infer_binary_expression(binary);
+                Type::Todo
```

Comment:
I think actually supporting this would take only like three lines here, plus some tests 😆 maybe a good task for @charliermarsh

Reply:
I can do it!